### PR TITLE
#348 Data Transfer Object design pattern

### PR DESCRIPTION
**Pull request by gopinath-langote**
Basic version of Data transfer object pattern implementation.

### CODE REVIEW

File: data-bus/pom.xml [hunk 1]
```diff
@@ -27,6 +27,10 @@
          xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
     <modelVersion>4.0.0</modelVersion>
+    <packaging>pom</packaging>
+    <modules>
+        <module>../data-transfer-object</module>
+    </modules>
```
Comment (iluwatar) [inline data-bus/pom.xml]: Why is Data Bus pattern affected? Should this change be reverted?

File: data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+   * @param args program argument.
+   */
+  public static void main(String[] args) {
+    List<CustomerDto> customers = new ArrayList<>();
+    CustomerDto customerOne = new CustomerDto("1", "Kelly", "Brown");
+    CustomerDto customerTwo = new CustomerDto("2", "Alfonso", "Bass");
+    customers.add(customerOne);
+    customers.add(customerTwo);
+
+    CustomerResource customerResource = new CustomerResource(customers);
```
Comment (iluwatar) [inline data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java]: This is customer server data initialization, right? Could we move to static block in `CustomerResource` class?

File: data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    System.out.println("Adding customer three}");
+    CustomerDto customerThree = new CustomerDto("3", "Lynda", "Blair");
+    customerResource.save(customerThree);
+    allCustomers = customerResource.getAllCustomers();
+    printCustomerDetails(allCustomers);
+  }
+
+  private static void printCustomerDetails(List<CustomerDto> allCustomers) {
+    allCustomers.forEach(customer -> System.out.println(customer.getFirstName()));
```
Comment (iluwatar) [inline data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java]: Don't use `System.println`, instead add and use a logger as in other examples.

File: data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerDto.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * {@link CustomerDto} is a data transfer object POJO. Instead of sending individual information to client
+ * We can send related information together in POJO.
+ * <p>
+ * Dto will not have any business logic in it.
+ */
+public class CustomerDto {
+  private String id;
+  private String firstName;
+  private String lastName;
```
Comment (iluwatar) [inline data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerDto.java]: This looks like immutable class. The private members should be declared `final`.

File: data-bus/pom.xml (hunk 1 above)
Comment (gopinath-langote) [inline data-bus/pom.xml]: @iluwatar 👍 . 
Should be reverted. I will do that.

File: data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java (hunk 2 above)
Comment (gopinath-langote) [inline data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java]: @iluwatar To keep is simple yes. We can do that. Rather that looks more appropriate. 
Thanks for suggestions. Let me do that.

File: data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java (hunk 3 above)
Comment (gopinath-langote) [inline data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java]: @iluwatar 👍 Ok. Make sense. I'll do that.

File: data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerDto.java (hunk 4 above)
Comment (gopinath-langote) [inline data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerDto.java]: @iluwatar : Sure. Thanks for pointing this out. I'll to the same.

**Comment by gopinath-langote** [comment]
@iluwatar : Added missing class diagram. 

Thanks.

**Comment by iluwatar** [comment]
@gopinath-langote looks good now. Thanks for the contribution! :+1:

**Pull request merged**